src/aat/java/uk/gov/hmcts/ccd/fta/steps/BackEndFunctionalTestScenarioPlayer.java

```diff
@@ -0,0 +1,270 @@
+        }
+
+        scenarioContext.setTheResponse(responseData);
+    }
+
+    @Override
+    @Then("a positive response is received")
+    public void verifyThatAPositiveResponseWasReceived() {
+        int responseCode = scenarioContext.getTheResponse().getResponseCode();
+        String errorMessage = "Response code '" + responseCode + "' is not a success code";
+        Assert.assertEquals(errorMessage, 2, responseCode / 100);
+        scenario.write("" + scenarioContext.getTheResponse().getResponseCode());
+    }
+
+    @Override
+    @Then("a negative response is received")
+    public void verifyThatANegativeResponseWasReceived() {
+        int responseCode = scenarioContext.getTheResponse().getResponseCode();
+        String errorMessage = "Response code '" + responseCode + "' is not a negative code";
+        Assert.assertNotEquals(errorMessage, 2, responseCode / 100);
+        scenario.write("" + scenarioContext.getTheResponse().getResponseCode());
+    }
+
+    @Override
+    @Then("the response has all the details as expected")
+    public void verifyThatTheResponseHasAllTheDetailsAsExpected() throws IOException {
+        ResponseData expectedResponse = scenarioContext.getTestData().getExpectedResponse();
+        ResponseData actualResponse = scenarioContext.getTheResponse();
+        Map<String, List> issues = new HashMap<>();
+
+        if (actualResponse.getResponseCode() != expectedResponse.getResponseCode()) {
+            issues.put("responseCode", Collections.singletonList("Response code mismatch, expected: "
+                + expectedResponse.getResponseCode() + ", actual: " + actualResponse.getResponseCode()));
+        }
+
+        MapVerificationResult headerVerification = MapVerifier.verifyMap(expectedResponse.getHeaders(),
+            actualResponse.getHeaders(), 1);
+        if (!headerVerification.isVerified()) {
+            issues.put("headers", headerVerification.getAllIssues());
+        }
+
+        MapVerificationResult bodyVerification = MapVerifier.verifyMap(expectedResponse.getBody(),
+            actualResponse.getBody(), 20);
+        if (!bodyVerification.isVerified()) {
+            issues.put("body", bodyVerification.getAllIssues());
+        }
+
+        scenario.write(JsonUtils.getPrettyJsonFromObject(scenarioContext.getTheResponse()));
+
+        if (issues.get("responseCode") != null || issues.get("headers") != null || issues.get("body") != null) {
+            Assert.fail("Response failures: " + JsonUtils.getPrettyJsonFromObject(issues));
+        }
+    }
+
+    @Override
+    @Then("the response [{}]")
+    public void verifyTheResponseInTheContextWithAParticularSpecification(String responseSpecification) {
+        String errorMessage = "Test data does not confirm it meets the specification about the response: "
+                + responseSpecification;
+        boolean check = scenarioContext.getTestData().meetsSpec(responseSpecification);
+        Assert.assertTrue(errorMessage, check);
```

Comment:
Maybe some benefit to printing the test data in the error too so can have an idea of expected/actual straight from the report rather than having to find the data file

Reply:
the missing specification is given in the error message already